cmd/kubeadm/app/cmd/phases/bootstraptoken.go

```diff
@@ -337,23 +331,19 @@ func createBootstrapToken(client clientset.Interface, cfgPath string, cfg *kubea
 	internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, cfg)
 	kubeadmutil.CheckErr(err)
 
-	// Load the CA certificate from so we can pin its public key
-	caCert, err := pkiutil.TryLoadCertFromDisk(internalcfg.CertificatesDir, kubeadmconstants.CACertAndKeyBaseName)
-	if err != nil {
-		return fmt.Errorf("error loading ca cert from disk: %v", err)
-	}
-
 	// Creates or updates the token
 	if err := node.UpdateOrCreateToken(client, internalcfg.Token, false, internalcfg.TokenTTL.Duration, usages, extraGroups, description); err != nil {
 		return err
 	}
 
 	fmt.Println("[bootstraptoken] Bootstrap token Created")
-	if skipTokenPrint {
-		internalcfg.Token = "{token}"
-	}
 	fmt.Println("[bootstraptoken] You can now join any number of machines by running:")
-	fmt.Printf("[bootstraptoken]   kubeadm join {master} --token %s --discovery-token-ca-cert-hash %s \n", internalcfg.Token, pubkeypin.Hash(caCert))
+
+	joinCommand, err := cmdutil.GetJoinCommand(kubeConfigFile, internalcfg.Token, skipTokenPrint)
+	if err != nil {
```

Comment:
So I generally like everything except what happens in this failure condition.  From what I grok, if you fail you don't see the full kubeadm join flow.

Reply:
getJoinCommand could fail only if the admin.conf is not available/is invalid, but in that case the operation most probably will fail long before reaching this point.

However, if all the preceding steps completes but getJoinCommand fails for any reason, in the current implementation you don't see printed the `kubeadm join` command to use for joining nodes, but instead you see printed the message  "failed to get join command: %error".

Is that  acceptable for you? If not what do you suggest? To improve the error message? How? Printing a static  `kubeadm join` command without with fixed values instead of variables read from admin.conf e.g. kubeadm join {api-server-address} ....?

Reply:
I guess I'm ok with the change as the user can just hit the up-arrow.